### ISSUE TITLE
How to handle if sample melgram's shape is smaller so that I cant broadcast the next file.?

### ISSUE THREAD

**Issue by Navdevl** [drscotthawley/panotti#1]
`ERROR: mel_dims =  (1, 1, 96, 156) , melgram.shape =  (1, 1, 96, 147)`

I can't pad it to fix this. Do you got any util function to fix this ?

**Comment by Navdevl** [comment]
Can you write a util function that could take the file with largest length and zero pad all other files accordingly by choosing that one as sample file rather than the first one. ?

**Comment by drscotthawley** [comment]
@Navdevl I've been considering such changes to make the code more 'accessible'.  My bias as a scientist had been to keep the dataset as 'clean' as possible, but after consulting with a local CS prof we decided that auto-cleaning measures like you describe would help.  So, I'll work on this and will test kasbah's pull request....

**Comment by drscotthawley** [comment]
@Navdevl and @kasbah  I made the merge, and yet also found that just going based on the longest file as in @kasbah 's PR could generate a huge waste of resources if most files were short but one happened to be long.  Thus I added a --dur tag in which to set the maximum allowed duration.  Check out the new "Sorting Hat" module at https://github.com/drscotthawley/panotti/tree/master/sorting-hat, where we set this to 4.0s.
